src/main/java/com/lpvs/service/LPVSQueueService.java

```diff
@@ -258,20 +230,35 @@ public void processWebHook(LPVSQueue webhookConfig) {
                 log.debug("Creating comment");
                 gitHubService.commentResults(webhookConfig, files, detectedConflicts, pullRequest);
                 log.debug("Results posted on GitHub");
+                delete(webhookConfig);
             } else {
                 log.warn("Files are not found. Probably pull request is not exists.");
                 gitHubService.commentResults(webhookConfig, null, null, pullRequest);
-                delete(webhookConfig);
                 throw new Exception(
                         "Files are not found. Probably pull request does not exist. Terminating.");
             }
-            delete(webhookConfig);
         } catch (Exception | Error e) {
             pullRequest.setStatus(LPVSPullRequestStatus.INTERNAL_ERROR.toString());
             pullRequest = lpvsPullRequestRepository.saveAndFlush(pullRequest);
             log.error("Can't authorize commentResults() " + e);
             e.printStackTrace();
-            delete(webhookConfig);
+            int currentAttempts = webhookConfig.getAttempts();
+            if (currentAttempts < maxAttempts) {
+                webhookConfig.setAttempts(currentAttempts + 1);
+                try {
+                    addFirst(webhookConfig);
+                } catch (InterruptedException e1) {
+                    log.warn("Failed to update Queue element");
+                }
+                queueRepository.save(webhookConfig);
+            } else {
+                log.warn(
+                        "Maximum amount of processing webhook reached for pull request: "
+                                + pullRequest.getId()
+                                + " "
+                                + pullRequest.getPullRequestUrl());
+                delete(webhookConfig);
```

Comment:
Before deleting webhook, we have to post results on GitHub:
```suggestion
                delete(webhookConfig);
                gitHubService.commentResults(webhookConfig, null, null, pullRequest);
```